.travis.yml

```diff
@@ -29,3 +29,8 @@ script:
 - if [ "$OS" = "linux" ]; then bash "$TRAVIS_BUILD_DIR"/tools/travis/ci-build-linux.sh; fi
 - if [ "$OS" = "windows" ]; then bash "$TRAVIS_BUILD_DIR"/tools/travis/ci-build-windows-ms.sh; fi
 - if [ "$OS" = "linux" -a "$TRAVIS_PULL_REQUEST" != "false" ]; then bash "$TRAVIS_BUILD_DIR"/tools/travis/pr-build.sh; fi
+- cd "$TRAVIS_BUILD_DIR"
```

Comment:
-  If Lua checking is mandatory for the linux build then IMO it should be called from `ci-build-linux.sh` and not by extra lines in the YML.
You haven't changed the `run-check.sh` on this PR but you should:
-  If `luacheck` is already on the path then the install components should be skipped.  This facilitates developers running the script when developing as well as during Travis QA runs.
-  Ditto if luarocks is already installed then the only the rocks install of luacheck should be done.  As I said in a previous comment we should not be running builds of Lua and Luarocks when they are already available as package addons in the `.travis,yml`

Reply:
Those additional commands lines are using `luac.cross` to check if the .lua files have no syntax errors and this check is used both in Linux and Windows build - that way also `luac.cross` binary can be tested on both platforms. 
I modified `run-luacheck.sh` to use pre-installed LuaRocks by default, but added an standalone install option just in case. Tested both options and everything seems to work fine (at least in docker environment).

It may be a good option to make this check before building a whole firmware or even better - check if commit changes only .lua files, only C stuff or both (using [TRAVIS_COMMIT_RANGE](https://docs.travis-ci.com/user/environment-variables/)) and run firmware build or luacheck only when necessary but this is out of scope of this PR.

Reply:
It might also make sense to have another job for luacheck, just like for windows. That way luacheck could run in parallel and would not require any extra waiting time.

Reply:
@HHHartmann I added script to run Luacheck on Windows and it seems to be working properly.

BTW I just realised that I misunderstood your comment and added Luacheck for windows instead of making it as a separate job. Sorry :smile: It runs much faster on Windows though (because only `luac.cross` is built in that enviroment) and the script itself can be used in normal Windows environment with Bash included in Git for Windows and wget.

Reply:
Does luacheck for linux yield difrfrent results than for windoes? If not maybe the windoes part would be enough. But then the linux version is not tested and starts rotting.

Reply:
`luacheck` for both Windows and GNU/Linux should give the exact same output, so it's not really necessary to run it in both environments. The only advantage of using Windows is that it's faster to check because there is official pre-compiled binary of `luacheck` with built-in Lua 5.3 and other libraries + Travis can cache the binary itself. A good idea would be to look into Travis CI docs and add separate jobs for changes in Lua code and changes in C code, but that's out-of-scope for this PR.

Reply:
So should we use the windows version only here to save time and add the linux version only to a release build to detect when it is broken?
@TerryE Terry what is your opinion on this?